felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -826,13 +877,24 @@ func (m *bpfEndpointManager) updateWEPsInDataplane() {
 // applyPolicy actually applies the policy to the given workload.
 func (m *bpfEndpointManager) applyPolicy(ifaceName string) error {
 	startTime := time.Now()
+	isReady := false
+
+	// Update the readiness state when exiting
+	defer func() {
+		m.ifacesLock.Lock()
```

Comment:
Just spotted an issue here; if the code inside the withIface block below ever panics then this Lock() will deadlock and we'll never get the stack trace.  Suggest moving the block below into a function so the Unlock can be deferred.  Will probably never happen but trying to debug it would be tough if it did!